src/avx2-64bit-qsort.hpp

```diff
@@ -98,12 +103,19 @@ struct avx2_vector<int64_t> {
     static reg_t
     mask_i64gather(reg_t src, opmask_t mask, __m256i index, void const *base)
     {
-        return _mm256_mask_i64gather_epi64(src, base, index, mask, scale);
+        return _mm256_mask_i64gather_epi64(
+                src, (const long long int *)base, index, mask, scale);
```

Comment:
I'm guessing this was never used before and hence never failed at compile time?

Reply:
Yeah, since it was a template and was never instantiated, it never mattered if it was correct or not.